### DISCUSSION TITLE
Firmware Git Hash

### DISCUSSION THREAD

**Discussion by Poofjunior** [harp-tech/protocol#16] [Open-ended]
Hi folks,

Is there any reason Harp stores firmware revision numbers instead of a git hash?

The shortened git hash is
* 8 bytes long
* pins the exact version of the code and can be restored with `git checkout <hash>`
* can be generated at compile-time and inserted into the code with a `#define` macro.

Perhaps this is a change that could be worth considering in the 32-bit protocol?

**Comment by bruno-f-cruz** [comment]
I might be missing the full scope of the suggestion, but wouldn't this be incompatible with semantic versioning? It would make it rather hard to keep tracking of Hw <-> Fw <-> Interface cross-compatibility since all IDs would be hashed.

**Reply by bruno-f-cruz** [reply]
While I still think the FW version should be saved under a SemVer format, I still think it is worth thinking about how to have development suffixes. For instance, NuGet Versions allow:
`1.2.3-preview555`
Following @Poofjunior suggestion, this suffix could be anything, including a github hash. It could thus be interesting to allow an additional core register to keep track of this value.

**Comment by Poofjunior** [comment]
Oof; I should've clarified this earlier. What I would love would be a core register for a git hash of the device's **app** firmware, not for the Harp Core firmware. Storing the git hash of the Harp core doesn't actually give us a way of pinpointing exactly what firmware is running on the device. Saving the app firmware git hash does though.
